libs/cardano-ledger-binary/src/Cardano/Ledger/Binary/Decoding/Decoder.hs

```diff
@@ -915,16 +957,15 @@ decodeMapEnforceNoDuplicates decodeKey decodeValueFor = do
       if shouldStop
         then pure acc
         else do
-          (k, v) <- decodeInlinedPair
-          when (k `Map.member` acc) $ fail "Duplicate key detected in the Map"
-          loop condition nextStep (Map.insert k v acc)
-    decodeInlinedPair = do
-      !key <- decodeKey
-      !value <- decodeValueFor key
-      pure (key, value)
+          !key <- decodeKey
+          !value <- decodeValueFor key
```

Comment:
is this just a style change, or is there something substantive?

Reply:
Instead of adding an `INLINE` pragma for the function to be inlined, I decided to literally inline the function. Also, note that allocating of a tuple is no longer necessary.

Reply:
ah!